sktime/forecasting/dynamicfactor.py

```diff
@@ -0,0 +1,411 @@
+        figsize : tuple , optional
+            If a figure is created, this argument allows specifying a size.
+            The tuple is (width, height).
+        auto_ylims : bool , optional
+            If True, adjusts automatically the y-axis limits to ACF values.
+        bartlett_confint : bool , default = True
+            Confidence intervals for ACF values are generally placed at 2
+            standard errors around r_k. The formula used for standard error
+            depends upon the situation. If the autocorrelations are being
+            used to test for randomness of residuals as part of the ARIMA routine,
+            the standard errors are determined assuming the residuals are white noise.
+            The approximate formula for any lag is that standard error
+            of each r_k = 1/sqrt(N).For the ACF of raw data, the standard error at
+            a lag k is found as if the right model was an MA(k-1).
+            This allows the possible interpretation that if all autocorrelations
+            past a certain lag are within the limits,
+            the model might be an MA of order defined by
+            the last significant autocorrelation.
+            In this case, a moving average model is assumed for the data
+            and the standard errors for the confidence intervals should be generated
+             using Bartlett’s formula.
+        acf_kwargs : dict , optional
+            Optional dictionary of keyword arguments that are directly
+            passed on to the correlogram Matplotlib plot produced by plot_acf().
+
+
+        Returns
+        -------
+        Figure
+            Figure instance with diagonistic plots.
+        """
+        self._fitted_forecaster.plot_diagonistics(
+            variable=variable,
+            lags=lags,
+            fig=fig,
+            figsize=figsize,
+            truncate_endog_names=truncate_endog_names,
+            auto_ylims=auto_ylims,
+            bartlett_confint=bartlett_confint,
+            acf_kwargs=acf_kwargs,
+        )
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str , default = "default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for forecasters.
+
+
+        Returns
+        -------
+        params :dict or list of dict , default = {}
+            arameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params
+        """
+        return super().get_test_params()
```

Comment:
why are you calling `super` here?

Reply:
if you wanted to do this, you can simply delete the function. "inheritance" ensures that, if a function is not implemented, the parent's function is called.

Although you probably don't want to leave it empty?

Reply:
Yes I didn't want to leave it empty ; also I thought the get_test_params method is defined in statsmodeladapter class so it would return parameters for any statsmodel forcaster but now I discovered the method is  from baseobject class in sktime/base/_base.py . I was not aware I needed to return specific params . Thanks for helping me out .